src/snowcli/plugins/nativeapp/run_processor.py

```diff
@@ -341,6 +347,12 @@ def upgrade_app(
                     """
                     )
                 )
+
+                # ensure debug_mode is up-to-date
```

Comment:
could you please elaborate why this is needed? Is there anything in between that might be reset the debug_mode flag?

Reply:
Yes, a user could have run `snow app run` initially, which creates the application in debug mode. Then they play around with it in Snowsight and may accidentally or intentionally turn it off. 
Now when they upgrade, it is on the user to keep the value consistent in project definition file, and we will follow that and keep the debug mode as such.